core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1158,12 +1158,18 @@ public int getNumberOfPinnedFiles() {
   public void delete(AlluxioURI path, DeleteOptions options) throws IOException,
       FileDoesNotExistException, DirectoryNotEmptyException, InvalidPathException,
       AccessControlException {
+    List<Inode<?>> deletedInodes;
     Metrics.DELETE_PATHS_OPS.inc();
     try (JournalContext journalContext = createJournalContext();
          LockedInodePath inodePath = mInodeTree.lockFullInodePath(path, InodeTree.LockMode.WRITE)) {
       mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
       mMountTable.checkUnderWritableMountPoint(path);
-      deleteAndJournal(inodePath, options, journalContext);
+      deletedInodes = deleteAndJournal(inodePath, options, journalContext);
```

Comment:
Yes, but that is assuming the method does not have side-effects when an exception is thrown. It might be the case that a delete throws an exception, and "some" files have been deleted. This is the same reason why we no longer return the journal entry id.